tools/osv-linter/internal/checks/checks.go

```diff
@@ -0,0 +1,140 @@
+// Package checks defines and implements all checks and collections of checks.
+//
+// To add additional checks:
+// 1. define a new instance of `Check`
+// 2. add it to the `checks` array
+// 3. add it to the relevent collections defined in `checkCollections`
+//
+// To add additional collections of checks:
+// 1. add to the `checkCollections` array.
+package checks
+
+import (
+	"fmt"
+
+	"github.com/tidwall/gjson"
+)
+
+// A CheckCode is a unique code for a check.
+type CheckCode string
+
+// CheckError describes when a check fails.
+type CheckError struct {
+	Code    CheckCode
+	Message string
+}
+
+// Error returns the error message, including the code.
+func (ce *CheckError) Error() string {
+	return fmt.Sprintf("%s: %s", ce.Code, ce.Message)
+}
+
+// CodeString returns just the error code, as a string.
+func (ce *CheckError) CodeString() string {
+	return string(ce.Code)
+}
+
+// Checkers are for running a discrete checking function.
+type Checker interface {
+	CodeString() string
+	Name() string
+	Description() string
+	Run(*gjson.Result) []CheckError
+}
+
+// Check defines a single check.
+type Check struct {
```

Comment:
nit: with the interface defined, we don't need this struct anymore -- it's redundant and seems more like an implementation detail that should be omitted from the interface.  

We can just define checks (in individual files) like so:

```
type CheckIntroducedEventExists struct{}

func (c CheckIntroducedEventExists) Name() {
  return "blah"
}

func (c CheckIntroducedEventExists) CodeString() {
  return "R0001"
}

func (c CheckIntroducedEventExists) Run(..) []CheckError {
}

```

Reply:
I think this is obsolete with subsequent changes